packages/engine/src/ecs/functions/SystemFunctions.ts

```diff
@@ -30,26 +30,37 @@ export type SystemInstanceType = {
 
 export const initSystems = async (world: World, systemModulesToLoad: SystemModuleType<any>[]) => {
   const loadSystemInjection = async (s: SystemFactoryType<any>) => {
-    const system = await s.systemModule.default(world, s.args)
     const name = s.systemModule.default.name
-    return {
-      name,
-      type: s.type,
-      sceneSystem: s.sceneSystem,
-      execute: () => {
-        const start = nowMilliseconds()
-        try {
-          system()
-        } catch (e) {
-          console.error(e)
-        }
-        const end = nowMilliseconds()
-        const duration = end - start
-        if (duration > 10) {
-          console.warn(`Long system execution detected. System: ${name} \n Duration: ${duration}`)
+    try {
+      const system = await s.systemModule.default(world, s.args)
+      return {
+        name,
+        type: s.type,
+        sceneSystem: s.sceneSystem,
+        execute: () => {
+          const start = nowMilliseconds()
+          try {
+            system()
+          } catch (e) {
+            console.error(e)
+          }
+          const end = nowMilliseconds()
+          const duration = end - start
+          if (duration > 10) {
+            console.warn(`Long system execution detected. System: ${name} \n Duration: ${duration}`)
+          }
         }
-      }
-    } as SystemInstanceType
+      } as SystemInstanceType
+    } catch (e) {
+      console.error(`System ${name} failed to initialize! `)
+      console.error(e)
+      return {
```

Comment:
should we return an empty system? or return null and filter it out?

Reply:
We can filter it out, sure